src/HooglePlus/FilterTest.hs

```diff
@@ -0,0 +1,194 @@
+module HooglePlus.FilterTest (runChecks, runChecks') where
+
+import Language.Haskell.Interpreter
+import Language.Haskell.Exts.Parser
+import Text.Printf
+import Control.Exception
+import Data.List
+import Language.Haskell.Exts.SrcLoc
+import Language.Haskell.Exts.Syntax
+import Language.Haskell.Exts.Pretty
+import Test.QuickCheck
+import qualified Data.Map as Map hiding (map, foldr)
+import qualified Data.Set as Set hiding (map)
+import System.Timeout
+import Control.Monad
+import Data.Maybe
+import Data.Typeable
+
+import HooglePlus.Utils
+import Types.Environment
+import Types.Program
+import Types.Type hiding (typeOf)
+import Synquid.Type
+
+excludeFunctions :: [String]
+excludeFunctions = ["GHC.List.iterate"]
+
+rejectInfiniteFunctions :: Bool
+rejectInfiniteFunctions = True
+
+data ArgumentType = Concrete String
+                  | Polymorphic String
+                  | ArgTypeList ArgumentType
+                    deriving (Eq)
+
+instance Show ArgumentType where
+  show (Concrete name) = name
+  show (Polymorphic _) = "Int"
+  show (ArgTypeList sub) = printf "[%s]" (show sub)
+
+data NotSupportedException = NotSupportedException String
+  deriving (Show, Typeable)
+
+instance Exception NotSupportedException
+
+type TypeEnv = [(ArgumentType, String)]
+
+parseTypeString :: String -> (TypeEnv, [ArgumentType], ArgumentType)
+parseTypeString input = buildRet [] [] value
+  where
+    (ParseOk value) = parseType input
+
+    argName (TyVar _ (Ident _ name)) = Polymorphic name
+    argName (TyCon _ (UnQual _ (Ident _ name))) = Concrete name
+    argName (TyList _ arg) = ArgTypeList (argName arg)
+    argName (TyParen _ t) = argName t
+    argName other = throw $ NotSupportedException ("Not able to handle " ++ show other)
+
+    buildRet typeEnv argList (TyForall _ _ (Just ctx) t) = buildRet typeEnv' argList t
+      where typeEnv' = typeEnv ++ buildEnv typeEnv ctx
+    buildRet typeEnv argList (TyFun _ typeArg typeRet) = buildRet typeEnv argList' typeRet
+      where argList' = argList ++ [argName typeArg]
+    buildRet typeEnv argList typeRet = (typeEnv, argList, argName typeRet)
+
+    extractQualified (ClassA _ (UnQual _ (Ident _ name)) var) =
+      map (\x -> (argName x, name)) var
+    extractQualified (ParenA _ qual) = extractQualified qual
+    extractQualified other = throw $ NotSupportedException ("Not able to extract " ++ show other)
+    buildEnv typeEnv (CxSingle _ item) = typeEnv ++ extractQualified item
+    buildEnv typeEnv (CxTuple _ list) = foldr ((++) . extractQualified) typeEnv list
+
+generateRandomValue :: String -> IO String
+generateRandomValue typeName = do
+  result <- runInterpreter $ do {
+    setImports ["Prelude", "Test.QuickCheck"];
+
+    act <- interpret ("do {\
+    \it <- generate (arbitrary :: Gen (" ++ typeName ++ "));\
```

Comment:
It doesn't look like this will work with bytestring, it has to be imported. The environment passed in from GHCChecker should contain a list of all modules you should import, like how we do for demand analysis.